pkg/controller/projects/projects/project.go

```diff
@@ -149,6 +158,52 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 	if !ok {
 		return managed.ExternalCreation{}, errors.New(errNotProject)
 	}
+	// Line 161 -> 206 will be export to an util function
+	keySecretRef := cr.Spec.ForProvider.ImportUrlSecretRef
+	importUrl := cr.Spec.ForProvider.ImportURL
+
+	// User has provided secret for the import
+	if keySecretRef != nil && importUrl != nil {
+		// Retrieve secret from k8s
+		namespacedName := types.NamespacedName{
+			Namespace: keySecretRef.Namespace,
+			Name:      keySecretRef.Name,
+		}
+
+		secret := &corev1.Secret{}
+
+		err := e.kube.Get(ctx, namespacedName, secret)
+
+		if err != nil {
+			return managed.ExternalCreation{}, errors.New(errNoImportSecret)
+		}
+
+		// Obtain the password from the secret.
+		password := string(secret.Data[keySecretRef.PasswordKey])
```

Comment:
Check if key is present in Secret struct before doing anything with it.
Also please make sure empty string "" is handled properly (nil vs "").